app/grandchallenge/participants/forms.py

```diff
@@ -1,8 +1,86 @@
+from django.core.exceptions import ValidationError
+from django.db.transaction import on_commit
 from django.forms import HiddenInput, ModelForm, TextInput
 
 from grandchallenge.core.forms import SaveFormInitMixin
 from grandchallenge.core.widgets import JSONEditorWidget
-from grandchallenge.participants.models import RegistrationQuestion
+from grandchallenge.participants.form_fields import RegistrationQuestionField
+from grandchallenge.participants.models import (
+    RegistrationQuestion,
+    RegistrationQuestionAnswer,
+    RegistrationRequest,
+)
+
+
+class RegistrationRequestForm(ModelForm):
+    class Meta:
+        model = RegistrationRequest
+        fields = (
+            "user",
+            "challenge",
+        )
+        widgets = {
+            "user": HiddenInput(),
+            "challenge": HiddenInput(),
+        }
+
+    def __init__(self, *args, challenge, user, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user"].initial = user
+        self.fields["user"].disabled = True
+        self.fields["challenge"].initial = challenge
+        self.fields["challenge"].disabled = True
+
+        self._registration_questions = RegistrationQuestion.objects.filter(
+            challenge=challenge
+        )
+
+        for q in self._registration_questions:
+            self.fields[str(q.pk)] = RegistrationQuestionField(
```

Comment:
Should we have a better field name than just the pk? `registration_question_{q.pk}` maybe? Could also make it a model property for re-use if necessary.

Reply:
N/A